tests/test_utils.py

```diff
@@ -833,3 +833,12 @@ def test_non_operations_in_list(self, arg):
             ValueError, match="The given operation_list does not only contain Operations"
         ):
             pu.inv(arg)
+
+    def test_warning(self):
+        """Test that the warning is generated."""
+
+        with pytest.warns(
+            UserWarning,
+            match="Use of qml.inv() is deprecated and should be replaced with qml.adjoint().",
```

Comment:
It seems from the test log, that it is not properly matching this warning:
```
 _____________________________ TestInv.test_warning _____________________________
Traceback (most recent call last):
  File "/home/runner/work/pennylane/pennylane/tests/test_utils.py", line 844, in test_warning
    qml.inv(qml.Hadamard(wires=[0]))
  File "/opt/hostedtoolcache/Python/3.9.5/x64/lib/python3.9/site-packages/_pytest/recwarn.py", line 288, in __exit__
    fail(
  File "/opt/hostedtoolcache/Python/3.9.5/x64/lib/python3.9/site-packages/_pytest/outcomes.py", line 153, in fail
    raise Failed(msg=msg, pytrace=pytrace)
Failed: DID NOT WARN. No warnings of type (<class 'UserWarning'>,) matching ('Use of qml.inv() is deprecated and should be replaced with qml.adjoint().') was emitted. The list of emitted warnings is: [UserWarning('Use of qml.inv() is deprecated and should be replaced with qml.adjoint().')].
```

This is because `pytest.warns` uses [regex](https://en.wikipedia.org/wiki/Regex) for matching errors and warnings, and the `.` and `(` ,`)` characters have special meaning in regex!

We can fix this by escaping these characters, to make sure they match periods and parenthesis respectively.

 ```suggestion
            match=r"Use of qml\.inv\(\) is deprecated and should be replaced with qml\.adjoint\(\)\.",
```